src/vertex_array_object.rs

```diff
@@ -286,6 +286,18 @@ impl VertexArrayObject {
             }
         }
 
+        // checking for duplicate attribute locations
+        for &(_, ref bindings, _, _, _) in vertex_buffers {
+            for i in 0..bindings.len() {
```

Comment:
```suggestion
            for (i, bi) in bindings.iter().enumerate() {
```
you can check iterators here, then you don't need to index below

Reply:
true!